MechanicsCore/src/main/java/me/deecaad/core/mechanics/defaultmechanics/MessageMechanic.java

```diff
@@ -102,14 +101,18 @@ private void send(Player player, CastData cast) {
         MiniMessage PARSER = MechanicsCore.getPlugin().message;
         Audience audience = MechanicsCore.getPlugin().adventure.player(player);
 
-        Component chat = chatStr == null ? null : PARSER.deserialize(placeholders(chatStr, player, cast));
-        Component actionBar = actionBarStr == null ? null : PARSER.deserialize(placeholders(actionBarStr, player, cast));
+        String itemTitle = cast.getItemTitle();
+        ItemStack itemStack = cast.getItemStack();
+        Map<String, String> tempPlaceholders = cast.getTempPlaceholders();
 
-        Component titleComponent = titleStr == null ? Component.empty() : PARSER.deserialize(placeholders(titleStr, player, cast));
-        Component subtitleComponent = subtitleStr == null ? Component.empty() : PARSER.deserialize(placeholders(subtitleStr, player, cast));
+        Component chat = chatStr == null ? null : PARSER.deserialize(PlaceholderAPI.applyPlaceholders(chatStr, player, itemStack, itemTitle, null, tempPlaceholders));
```

Comment:
quick glance at the code looks like it's possible for `PlaceHolderAPI` to skip using the temp placeholders (if there are no registered placeholders. Does this still work if placeholderapi is not run on the server?)